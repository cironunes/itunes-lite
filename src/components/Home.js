import React, { Component } from 'react';

import { AlbumSearchForm, AlbumList } from './album/';
import { searchAlbumByTerm } from '../services/ItunesService';

export class Home extends Component {

  state = {
    searchTerm: 'Iron Maiden',
    collections: []
  }

  componentDidMount() {
    searchAlbumByTerm(this.state.searchTerm)
      .then(res => res.json())
      .then(({ results }) => this.setState({ collections: results }));
  }

  handleInputChange = (evt) => {
    evt.persist();
    this._inputTimeout && clearTimeout(this._inputTimeout);
    this._inputTimeout = setTimeout(() => {
      this.setState({ searchTerm: evt.target.value })
      searchAlbumByTerm(evt.target.value)
        .then(res => res.json())
        .then(({ results }) => this.setState({collections: results}))
    }, 300);
  }


  render() {
    return (
      <div>
        <AlbumSearchForm handleInputChange={this.handleInputChange} searchTerm={this.state.searchTerm} />
        {this.state.collections && <AlbumList collections={this.state.collections} />}
      </div>
    );
  }
}
