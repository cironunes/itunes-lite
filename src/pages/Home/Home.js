import React from 'react';
import AlbumSearchForm from './components/AlbumSearchForm';
import AlbumListContainer from './components/AlbumList/AlbumListContainer';

const Home = () => (
  <div>
    <AlbumSearchForm />
    <AlbumListContainer />
  </div>
);

export default Home;
