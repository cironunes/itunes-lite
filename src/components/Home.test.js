import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { Home } from './Home';

const json = { json: () => ({ results: [
{ collectionId: 400, collectionName: 'foo', artworkUrl100: 'foo', artistName: 'foo', releaseDate: 'foo'  }
] }) };
jest.mock('react-router-dom', () => ({ Link: 'Link' }))
fetch = jest.fn().mockImplementation(() => new Promise(resolve => resolve(json)));

describe('RoutedComponent: Home', () => {
  test('get albuns when the component mounts', async () => {
    const component = await renderer.create(
      <Home />
    );

    let tree = await component.toJSON();
    expect(tree).toMatchSnapshot();

    component.getInstance().componentDidMount();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // component.getInstance().handleInputChange({ persist: () => {} });
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
