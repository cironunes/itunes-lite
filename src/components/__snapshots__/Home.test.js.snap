// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutedComponent: Home get albuns when the component mounts 1`] = `
<div>
  <form
    className="App-form"
  >
    <h2
      className="App-subtitle"
    >
      Have fun!
    </h2>
    <p
      className="App-form__hint"
    >
      Find your favorite albuns and get to know more about them
    </p>
    <div
      className="SearchForm"
    >
      <input
        autoFocus={true}
        className="App-form__field"
        defaultValue="Iron Maiden"
        onChange={[Function]}
        placeholder="search for albuns"
        type="search"
      />
    </div>
  </form>
  <ul
    className="Album-List"
  />
</div>
`;

exports[`RoutedComponent: Home get albuns when the component mounts 2`] = `
<div>
  <form
    className="App-form"
  >
    <h2
      className="App-subtitle"
    >
      Have fun!
    </h2>
    <p
      className="App-form__hint"
    >
      Find your favorite albuns and get to know more about them
    </p>
    <div
      className="SearchForm"
    >
      <input
        autoFocus={true}
        className="App-form__field"
        defaultValue="Iron Maiden"
        onChange={[Function]}
        placeholder="search for albuns"
        type="search"
      />
    </div>
  </form>
  <ul
    className="Album-List"
  >
    <li>
      <Link
        className="Card undefined"
        to="/album/400"
      >
        <img
          alt=""
          className="Media-Cover"
          src="foo"
        />
        <h1
          className="Media-Text Media-Title"
        >
          foo
        </h1>
        <h2
          className="Media-Text Media-Author ellipsis"
        >
          foo
        </h2>
        <span
          className="Media-Text Media-Date"
        >
          NaN
        </span>
      </Link>
    </li>
  </ul>
</div>
`;
